### PR TITLE
Add json format for metrics

### PR DESCRIPTION
**Pull request by rafaelldi**
This PR resolves what we've discussed in issue #1108. What do you think?

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1130) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1130) before we can accept your contribution.<br/><br/>:x: rafaelldi [sign now](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=1130)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/tye?pullRequest=1130) it.</sub>

**Comment by rafaelldi** [comment]
Hello! @philliphoff, could you take a look at my request if you have time?

File: src/Microsoft.Tye.Hosting/Model/V1/V1Metric.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Microsoft.Tye.Hosting.Model.V1
```
Comment (philliphoff) [inline src/Microsoft.Tye.Hosting/Model/V1/V1Metric.cs:1]: Nit: add the standard file license header to each new file.  (It looks like there are a couple of existing files that don't have it, which we should fix at some point, but is outside the scope of this PR.)

File: src/Microsoft.Tye.Hosting/TyeDashboardApi.cs [hunk 2]
```diff
@@ -262,6 +264,52 @@ private Task AllMetrics(HttpContext context)
...
+                return JsonSerializer.SerializeAsync(context.Response.Body, new
+                {
+                    message = $"Unknown service {name}"
+                },
+                _options);
+            }
+
+            var acceptHeader = context.Request.Headers[HeaderNames.Accept];
+
+            if (acceptHeader == MediaTypeNames.Application.Json)
```
Comment (philliphoff) [inline src/Microsoft.Tye.Hosting/TyeDashboardApi.cs:299]: Requests can declare multiple acceptable response formats; should we support those cases (i.e. tease out whether the preferred response is JSON and, if so, use that, else fallback to text (assuming that's even in the acceptable list))?

File: src/Microsoft.Tye.Hosting/Model/V1/V1Metric.cs (hunk 1 above)
Comment (rafaelldi) [reply src/Microsoft.Tye.Hosting/Model/V1/V1Metric.cs]: Fixed

File: src/Microsoft.Tye.Hosting/TyeDashboardApi.cs (hunk 2 above)
Comment (rafaelldi) [reply src/Microsoft.Tye.Hosting/TyeDashboardApi.cs:299]: It seems to me that the Prometheus format for metrics is more expected. The json format may be needed in rare cases, for example, as I now find it more convenient to parse a json string than to work with plain text in my plugin. Most users, in my opinion, get metrics through specialized services that know how to work with the Prometheus format. So I think it's more correct to give json only if explicitly asked for it, and in all other cases, respond in Prometheus format.
What do you think?

Comment (philliphoff) [reply src/Microsoft.Tye.Hosting/TyeDashboardApi.cs:299]: I'll buy that. :-)

**Review by philliphoff — APPROVED** [review]
Approved pending making the formatter and build happy.

**Comment by rafaelldi** [comment]
@philliphoff I'm not sure what I should do next. I've fixed formatting, but one workflow doesn't start automatically. Could you help me, please?

**Pull request merged**